src/Shared/CertificateGeneration/MacOSCertificateManager.cs

```diff
@@ -29,18 +31,18 @@ internal sealed class MacOSCertificateManager : CertificateManager
 
     // Delete a certificate with the specified SHA-256 (or SHA-1) hash {0} from keychain {1}.
     private const string MacOSDeleteCertificateCommandLine = "sudo";
-    private const string MacOSDeleteCertificateCommandLineArgumentsFormat = "security delete-certificate -Z {0} {1}";
+    private const string MacOSDeleteCertificateCommandLineArgumentsFormat = "security delete-certificate -Z {0} \"{1}\"";
 
     // Add a certificate to the per-user trust settings in the user keychain. The trust policy
     // for the certificate will be set to be always trusted for SSL and X.509 Basic Policy.
     // Note: This operation will require user authentication.
     private const string MacOSTrustCertificateCommandLine = "security";
-    private static readonly string MacOSTrustCertificateCommandLineArguments = $"add-trusted-cert -p basic -p ssl -k {MacOSUserKeychain} ";
+    private static readonly string MacOSTrustCertificateCommandLineArguments = $"add-trusted-cert -p basic -p ssl -k \"{MacOSUserKeychain}\" ";
 
     // Import a pkcs12 certificate into the user keychain using the unwrapping passphrase {1}, and
     // allow any application to access the imported key without warning.
     private const string MacOSAddCertificateToKeyChainCommandLine = "security";
-    private static readonly string MacOSAddCertificateToKeyChainCommandLineArgumentsFormat = "import {0} -k " + MacOSUserKeychain + " -t cert -f pkcs12 -P {1} -A";
+    private static readonly string MacOSAddCertificateToKeyChainCommandLineArgumentsFormat = "import {0} -k \"" + MacOSUserKeychain + "\" -t cert -f pkcs12 -P {1} -A";
```

Comment:
`{0}` is a path too, just a temporary file, right? Is it possible for the tmp dir to ever have a space? It feels better to quote all paths as a general rule.

Reply:
Under the hood, `$TMPDIR` is used to get the temp path, which macOS sets up in `/private/var` with a symlink in `/var/folders/...`. This should hold regardless of where the `$HOME` is.

But I guess in theory someone could redefine `$TMPDIR` so I'll go ahead and guard that path too.